util/design/secded_gen.py

```diff
@@ -589,45 +580,28 @@ def _inv_hamming_code(k, m):
     return _hamming_code(k, m)
 
 
-# n = total bits
-# k = data bits
-# m = parity bits
-# generate hamming code
-def _hamming_code(k, m):
-
-    n = k + m
-
+# Generate hamming code.
+# total_cnt = number of total bits
+# data_cnt = number of data bits
+# parity_cnt = number of parity bits
+@functools.lru_cache(maxsize=None)
```

Comment:
just double checking here, but was there a reason you wanted to not put a bound on this? the lru_cache just becomes a cache (that can grow without bound) when maxsize is set to `None`, according to the [docs](https://docs.python.org/3/library/functools.html#functools.lru_cache).

Reply:
Yes, to avoid the overhead of `lru_cache`. I don't expect this to be an issue for our use cases and I didn't notice any spikes while I was staring at `top`. 

P.S. Here is how I actually convinced myself: The size of our flash is 1 MiB, a flash word is 8 bytes, so roughly 130k entries. I don't know the exact storage format of the cache but let's assume 64 bytes per entry (8 bytes per arg and 8 args to be conservative), which is on the order of MBs.

Reply:
BTW, this applies to other functions that have a dataword argument, `_hamming_code()` is actually called for a handful of unique arg pairs since the args are bit counts.

Reply:
thanks for the explanation!